Sources/Passage/AuthorizationController/HostedAuthorizationController.swift

```diff
@@ -0,0 +1,320 @@
+                guard error == nil else {
+                    continuation.resume(throwing: error!)
+                    return
+                }
+                guard
+                    let self,
+                    let callbackURL,
+                    let (code, state): (String, String) = try? self
+                        .handleCallbackUrl(callbackUrl: callbackURL)
+                else {
+                    continuation.resume(throwing: HostedAuthorizationError.invalidHostedCallbackUrl)
+                    return
+                }
+                continuation.resume(returning: (code, state))
+            }
+            webAuthSession?.presentationContextProvider = self
+            webAuthSession?.prefersEphemeralWebBrowserSession = prefersEphemeralWebBrowserSession
+            webAuthSession?.start()
+        }
+    }
+    
+    @MainActor
+    internal func startWebAuthSafari() async throws -> (String, String) {
+        let url = try getHostedStartUrl()
+        safariViewController = SFSafariViewController(url: url)
+        let window = getAnchor()
+        guard
+            let safariViewController,
+            let appViewController = window.rootViewController
+        else {
+            throw HostedAuthorizationError.cannotAccessAppRootViewController
+        }
+        safariViewController.delegate = self
+        safariViewController.modalPresentationStyle = .pageSheet
+        appViewController.present(safariViewController, animated: true)
+        return try await withCheckedThrowingContinuation { [weak self] continuation in
+            self?.hostedContinuation = continuation
+        }
+    }
+    
+    internal func finishWebAuth(
+        authCode: String,
+        state: String,
+        clientSecret: String
+    ) async throws -> (AuthResult, String) {
+        let url = try getHostedFinishUrl(
+            authCode: authCode,
+            state: state,
+            clientSecret: clientSecret
+        )
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        let (data, response) = try await URLSession.shared.data(for: request)
+        guard let httpResponse = response as? HTTPURLResponse else {
+            throw HostedAuthorizationError.serverError
+        }
+        switch httpResponse.statusCode {
+        case 400: throw HostedAuthorizationError.invalidHostedTokenUrl
+        case 401: throw HostedAuthorizationError.unauthorized
+        case 500: throw HostedAuthorizationError.serverError
+        default: ()
+        }
+        let decoder = JSONDecoder()
+        guard let result = try? decoder.decode(HostedAuthResult.self, from: data) else {
+            throw HostedAuthorizationError.invalidHostedTokenUrl
+        }
+        let authResult = AuthResult(
+            authToken: result.accessToken,
+            redirectUrl: "",
+            refreshToken: result.refreshToken,
+            refreshTokenExpiration: nil
+        )
+        verifier = ""
```

Comment:
any reason that we have to set this to empty string?  For logout and finish hosted auth, we set it to a new value at the beginning anyways

Reply:
I just wanted to make sure it got erased from memory, but it's actually fine w/o this. I have updated it :)